### DISCUSSION TITLE
About adding the ability to edit file, label above value, all the displays and drag-and-drop

### DISCUSSION THREAD

**Discussion by gcarreno** [gcarreno/laz-JSON-Viewer#13] [Ideas]
Hey Marcus (@MFernstrom)

### It's a viewer

Since it's a viewer I want it to be added to the OS as the default `*.json` viewer.
On the `README.md` there's a way to do it for Linux(Ubuntu).
Craig Chapman (@chapmanworld) gave me the solution for Windows, just need to create a `regedit` file.
The goal is to have packages for Linux and an InnoSetup created Setup for Windows.
Also, since it's never gonna be more than a viewer, it will never have a menu so making it the default viewer from within, will never go.

### Edit File

The idea of a viewer came after I had a look at some offers on GitHub of viewers and editors and, after a bit of thought, realized that the quickest way to get a product out would be a simple viewer.

My ambitions for and editor are a bit high in terms of programming:
- Needs to have `TMemo`/`TSynEdit` linked with the `VST`.
- Needs to have the `VST` linked with `TMemo`/`TSynEdit`

You see, the hard part here is not the `TMemo`/`TSynEdit` or even the `VST` itself. The HUGE effort is on having it all synched, so when you edit the text, the tree reflects AND when you edit the tree the text reflects(without loosing caret position)

I'm still thinking about it and if enough people ask for the edit part I'll start a whole new project. I can't cram the edit part on this project.

### Label above value

I had it on an earlier version when I started to add all the displays, then to differentiate it from the displays, I removed it.

I still think you're right and something is needed. I think Ill replace the panel with a `TGroupBox`.

### All the displays

My whole idea for a viewer was to help me with different interpretations of Integers. And I think I crammed most of what you'll encounter in the wild.

The Date display is for Unix Timestamp, good hein!?

### Drag-and-Drop

I'm in the final stage of having it done. Just need to make sure all is good and I'll tag and release.

**Comment by MFernstrom** [comment]
If you want it to be a default viewer, I'd personally need it to be a viewer/editor. I work with JSON files a lot so my default has to be an editor.

**Reply by gcarreno** [reply]
That actually makes sense.

Looks like I have to finish this one quickly and then embark on the editor!!
